android_cobertura.md

```diff
@@ -0,0 +1,143 @@
+
+    doLast {
+        println "You can view the coverage report at:"
+        println "${buildDir}/reports/jacoco/testCoverage/html/index.html"
+    }
+}
+
+// Run and generates the report for the unit tests. The test report only includes the coverage data for
+// the presenter and utils classes since we generally don't unit test any Android related classes
+task unitTestCoverage(type: JacocoReport, dependsOn: 'testDebugUnitTest') {
+    sourceDirectories = coverageSourceDirs
+    classDirectories = fileTree(
+            dir: "${buildDir}/intermediates/classes/debug",
+            includes: ['**/*Presenter*',
+                       '**/utils/*',
+                       '**/drawing/**'
+                       // Add any additional testable classes here
+            ]
+    )
+    executionData = files("${buildDir}/jacoco/testDebugUnitTest.exec")
+
+    doLast {
+        println "You can view the coverage report at:"
+        println "${buildDir}/reports/jacoco/unitTestCoverage/html/index.html"
+    }
+
+    reports {
+        xml.enabled true
+        xml.destination "${buildDir}/reports/jacoco/unitTestCoverage/coverage.xml"
+    }
+}
+
+// Run and generates the report for the instrumentation test
+task instrumentationTestCoverage(type: JacocoReport, dependsOn: 'spoonDebugAndroidTest') {
+    sourceDirectories = coverageSourceDirs
+    classDirectories = coverageClassDirs
+
+    executionData = files(instrumentationTestCoverageData)
+
+    doLast {
+        println "You can view the coverage report at:"
+        println "${buildDir}/reports/jacoco/instrumentationTestCoverage/html/index.html"
+    }
+}
+
+android {
+    buildTypes {
+        debug {
+            testCoverageEnabled(project.hasProperty('coverage'))
+        }
+    }
+
+    testOptions {
+        unitTests.all {
+            jacoco {
+                // needed so that Robolectric tests are included in the results
+                includeNoLocationClasses = true
+            }
+        }
+    }
+}
+
+spoon {
+    codeCoverage = project.hasProperty('coverage')
+}
+```
+
+The main takeaway from this coverage.gradle is the addition of these lines within the `unitTestCoverage` task:
+```
+reports {
+    xml.enabled true
+    xml.destination "${buildDir}/reports/jacoco/unitTestCoverage/coverage.xml"
+}
+```
+This code snippet tells JaCoCo to output its results in XML format. This XML will then be converted to a format that Cobertura can read. `unitTestCoverage` will then be added to the Gradle script (see below) to perform the code coverage reporting.
+
+# Jenkins Build Phase
+These are the Build Steps to add to the Build Phase. **They must be added in the order written here!**
+
+## Invoke Gradle script
+Add this build step and choose `Use Gradle Wrapper`.
+
+Check `From Root Build Script Dir`.
+
+Under Tasks, please put: `clean testDebugUnitTest unitTestCoverage -Pcoverage assembleQa` **Note:** assembleQa is needed for Northeastern, but might not be needed for your Android project.
```

Comment:
testDebugUnitTest can be removed since it's included by unitTestCoverage. The -Pcoverage might not needed either with recent version of gradle, but I'll need to double check on that.